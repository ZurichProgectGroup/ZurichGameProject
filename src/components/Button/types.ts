import { ButtonHTMLAttributes, ReactNode } from 'react';

export const enum ButtonSize {
    Large,
    Small,
}

export const enum ButtonVariant {
    Transparent,
    Filled,
}

export type OwnProps = {
    className?: string,
    type?: 'button' | 'reset' | 'submit',
    variant?: ButtonVariant,
    size?: ButtonSize,
    children: ReactNode,
} & ButtonHTMLAttributes<HTMLButtonElement>;

export type Props = OwnProps;
