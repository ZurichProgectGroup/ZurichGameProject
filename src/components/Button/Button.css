.button {
    background: transparent;
    font-size: 16px;
    text-align: center;
    border: 1px solid rgb(var(--main));
    box-shadow: 0 5px 14px rgba(var(--main), 0.4), inset 0 -9px 24px 1px rgba(var(--main), 0.68);
    text-transform: uppercase;
    border-radius: 8px;
    padding: 14px;
    white-space: normal;
    overflow: hidden;
    text-overflow: ellipsis;
    color: rgb(var(--font-color));
    transition: all 0.2s;

    &:focus,
    &:hover {
        outline: none;
        border: 1px solid rgb(var(--focus));
        box-shadow: 0 5px 14px rgba(var(--focus), 0.4), inset 0 -9px 24px 1px rgba(var(--focus), 0.68);
    }
}
