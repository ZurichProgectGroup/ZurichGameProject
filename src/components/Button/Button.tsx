import React from 'react';
import './Button.css';
import { Props } from './types';

const Button: Props = (props) => {
    const {
        children,
        ...buttonProps
    } = props;

    return (
        <button type="button" {...buttonProps} className="button">
            {children}
        </button>
    );
};

export default Button;
