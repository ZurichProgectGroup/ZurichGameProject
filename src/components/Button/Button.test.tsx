import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Button from './Button';

it('renders correctly', () => {
    const { asFragment } = render(<Button>button</Button>);

    expect(asFragment()).toMatchSnapshot();
});

it('calls callback on click', () => {
    const mockCallback = jest.fn();

    const { getByText } = render(<Button onClick={mockCallback}>button</Button>);

    fireEvent.click(getByText('button'));

    expect(mockCallback).toHaveBeenCalledTimes(1);
});
