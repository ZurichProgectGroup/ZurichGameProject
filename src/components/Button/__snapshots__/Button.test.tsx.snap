// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <button
    class="button button_small button_transparent"
    type="button"
  >
    button
  </button>
</DocumentFragment>
`;
